gluepyter/tests/test_glue_session.py

```diff
@@ -61,6 +63,59 @@ def test__read_view_state(yglue_session):
     assert len(state) > 0
 
 
+def test_add_data(yglue_session):
+    def nested_compare(value1, value2):
+        if isinstance(value1, list):
+            if isinstance(value2, list):
+                if len(value1) == len(value2):
+                    for v1, v2 in zip(value1, value2):
+                        if not nested_compare(v1, v2):
+                            return False
+                    return True
+                else:
+                    return False
+            else:
+                return False
+        if isinstance(value1, dict):
+            if isinstance(value2, dict):
+                for k1, v1 in value1.items():
+                    if k1 in value2.keys():
+                        if not nested_compare(v1, value2[k1]):
+                            return False
+                    else:
+                        return False
+                return True
+            else:
+                return False
+        else:
+            return value1 == value2
```

Comment:
This function merits a standalone definition, maybe in the `glue_utils.py` for future usages.